### PR TITLE
Feature/lobby UI

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

File: spacerace-client/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+///
+/// # Arguments
+///
+/// * `client` - The HTTP client for sending requests.
+/// * `player_password` - The password of the current player.
+/// * `host` - The server host URL.
+async fn run_game_loop(
+    client: reqwest::Client,
+    player_password: String,
+    host: String,
```
Comment (mjs) [inline spacerace-client/src/main.rs:163]: `host_url` or `server_url` would be a better name. Same for the environment variable.

File: spacerace-client/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+                );
+                thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+            } else {
+                // Game is active or queued, allow user to control
+                thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+            }
+        } else {
+            debug!("No active game. Press 'q' to quit.");
+            thrust_rotation_from_input_with_timeout(Duration::from_millis(200))?
+        };
```
Comment (mjs) [inline spacerace-client/src/main.rs:202]: That's quite an expression :)

File: spacerace-client/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                    ship.velocity[0],
+                    ship.velocity[1],
+                    ship.orientation
+                );
+            }
+        }
+
+        stdout.flush()?;
+        // Throttle updates slightly to avoid high CPU usage
+        sleep(Duration::from_millis(100)).await;
```
Comment (mjs) [inline spacerace-client/src/main.rs:241]: Is 100ms short enough to allow for a reasonable response to keyboard events? Seems a bit long.

**Review by mjs — COMMENTED** [review]
Looks great. A few little questions and suggestions.

File: spacerace-client/src/main.rs (hunk 3 above)
Comment (hardbyte) [reply spacerace-client/src/main.rs:241]: Good point, I'll decrease to 30ms

**Pull request merged**